stempy/main.cpp

```diff
@@ -12,24 +64,98 @@ int main (int argc, char *argv[])
     return 1;
   }
 
-  stempy::StreamReader reader(argv[1]);
+  string inputPath = argv[1];
+
+  // check the input path
+  if (checkIfDirectory(inputPath))
+  {
+    cout << endl << "Input path " << inputPath << " is valid, starting processing" << endl;
+    // get all the files in this path
+    const vector<string> files = GetDirectoryFiles(inputPath);
+    int numFiles = files.size();
+    cout << numFiles << " files have been found" << endl;
 
-  int count = 0;
-  while (true) {
-    auto stream = reader.read();
+    // all the STEM images that would be summed into one final STEM image
+    vector<STEMImage> allPartialSTEMImages;
+
+    // information about processing
+    int width = 160;
+    int height = 160;
+    // -1 indicate the center of the image
+    int centerX = -1;
+    int centerY = -1;
+
+    for (int i = 0; i < numFiles; i++)
+    {
+      cout << "Processing " << i+1 << "/" << numFiles << ": " << files[i] << endl;
+      string curFile = files[i];
+      // read the data
+      stempy::StreamReader reader(inputPath+curFile);
+      auto stream = reader.read();
+
+      // version number needs to be 1, 2 or 3
+      if (stream.header.version == 0) 
+      {
+        break;
+      }
+      // some printing stuff about the input data
+      cout << "Version: " << stream.header.version << endl;
+      cout << "Images in block: " <<  stream.header.imagesInBlock << endl;
+      cout << "Rows: " << stream.header.frameHeight << endl;
+      cout << "Columns: " << stream.header.frameWidth << endl;
+      cout << "Diffraction image numbers: ";
+      for (auto n : stream.header.imageNumbers) {
+        cout << n << " ";
+      }
+      cout << endl;
+
+      // generate current partial STEM image
+      cout << "Generating partial STEM image" << endl;
+      
+      // information about how to process
+      vector<int> innerRadii(stream.header.imagesInBlock, 40);
+      vector<int> outerRadii(stream.header.imagesInBlock, 288);
+      
+      // create STEM 32 partial STEM images
+      vector<STEMImage> imagesOfBlock = createSTEMImages(reader.begin(), reader.end(), innerRadii, outerRadii, width, height, centerX, centerY);
```

Comment:
Correct me if I'm wrong, but it looks like this is producing several identical stem images? It might be faster to create one, and then copy it to get the rest of them, although you would have to make sure the `shared_ptr`'s inside get deep copied.

Reply:
Yeah you are correct, sorry I misunderstood what createSTEMImages() returned. Thank you very much!